dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/ZipFileArtifactNotifier.java

```diff
@@ -691,12 +702,18 @@ public void notifyEntryChange(
 
         boolean isUpdate = !updatedPaths.isEmpty();
         if ( addedPaths.isEmpty() && !isUpdate && removedPaths.isEmpty() ) {
-            return;
+            return; // Ignore completely null updates
         }
 
-        validateNotification(addedPaths, updatedPaths, removedPaths);
+        // The net is to allow updates alone or removals alone.
 
-        notifyAllListeners(isUpdate);
+        String validationMessage = validateNotification(addedPaths, updatedPaths, removedPaths);
+        if ( validationMessage != null ) {
+        	// TODO: Need to understand what is happening here.
```

Comment:
(same as before) should a debug trace go here?  otherwise the validationMessage won't be traced since that method is marked `@Trivial`